usage_analytics/analytics_func.py

```diff
@@ -0,0 +1,72 @@
+def get_calling_class(recursive=True):
+    """
+    Traverses the call stack to find and return the class name of the first
+    caller that has a 'self' variable in its frame (i.e., a method call).
+    If no such frame is found, returns 'StandaloneFunction'.
+    """
+
+    import inspect
```

Comment:
import here likely adds runtime overhead and error potential. If this method is used add inspect import to the top of the module.